parsers/uottawa.js

```diff
@@ -1,183 +1,77 @@
+var _semesterRegex = /([0-9]{2}\s[A-Z]{1}[a-z]{2,8})\sto\s([0-9]{2}\s[A-Z]{1}[a-z]{2,8})\s([0-9]{4})/;
 
-
-	var courseInfo = [];
-	var courseInfoMatch= [];
-	var days = [];
-	var re = /([A-Z]+\s\d{4}\s[A-Z]\s\w{3}\s(?:[(]\d[)]))(\d{1,2}[:]\d{2}[A-Z]*\s[-]\s\d{0,2}[:]\d{2}(?:(?:A|P)M)?)(\w.*\d)(\w.*)/;	//reg ex to find the course data
-	//  Output SEG 3102 A LEC (1)13:00 - 14:30LPR 155Some, StéphaneREGISTERED
-	//Example of the reg ex and the groups it matched
-	// group 1 = SEG 3102 A LEC (1)
-	// group 2 = 13:00 - 14:30
-	// group 3 = LPR 155
-	// group 4 = Some, StéphaneREGISTERED
-
-	// find where the class are, they are stored in table > #WEEKLY_SCHED_HTMLAREA > .SSSTEXTWEEKLY
-	$('#frameContents').find('.TimeTableRegularTime').each(function(i, elem) {
-	  days[i] = $(this).closest('table').find('.DeAcDataGridListHeader')[0].children[$(this).index()+1].children[0].data; //found on github, this find the cell header,which is the weekday
-	  courseInfo[i] = $(this).text(); //store the string data of the cell details
-	  courseInfoMatch[i] = re.exec(courseInfo[i]); //Match the details with the regualar exp
-	});
-
-	return [days, courseInfoMatch]
+/**
+ * Returns a timetable array consisting of class section objects which
+ * are parsed and generated from the raw uOttawa timetable text.
+ * @param {String} text Raw timetable text.
+ * @return {Object[]} Returns an array of JSON objects for each parsed class
+ *                    section.
+ */
+function parse(text) {
+  var timetable = [];
+  // Split lines by course code
+  var courses = text.split(_courseRegexLookAhead);
+  for (var i = 0; i < courses.length; i++) {
+    var courseText = courses[i];
+    var semester = _semesterRegex.exec(courseText);
+    var course = _courseRegex.exec(courseText);
+    if (!course || !semester) continue;
+    // Split lines by class sections
+    var classes = courseText.split(_timeRegexLookAhead);
+    for (var j = 0; j < classes.length; j++) {
+      var professor = _professorRegex.exec(classes[j]);
+      var time = _timeRegex.exec(classes[j]);
+      var where = _locationRegex.exec(classes[j]);
+      if (!professor || !time || !where) continue;
+      var sem_start = new Date(semester[1] + ' ' + semester[3]);
+      var sem_end = new Date(semester[2] + ' ' + semester[3]);
+      // Build object for this section and push it to the timetable array			
+      timetable.push({
+        'course_code_faculty': course[1],
+        'course_code_number': course[2],
+        'section': course[3],
+        'course_name': course[4],
+        'professor': professor[0],
+        'semester_start':  {
+          'year': sem_start.getFullYear(),
+          'month': sem_start.getMonth(),
+          'day': sem_start.getDate()
+        },
+        'semester_end': {
+          'year': sem_end.getFullYear(),
+          'month': sem_end.getMonth(),
+          'day': sem_end.getDate()
+        },
+        'where': where[0],
+        'day': time[1],
+        'start_time': {
```

Comment:
Are you going to account for date format or time format 12HR and 24HR like I did?